apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -31,42 +29,63 @@ const initialChatMessages: ChatCompletionMessage[] = [
 
 const initialState: AITutorState = {
   level: undefined,
+  scriptId: undefined,
   aiResponse: '',
   isWaitingForAIResponse: false,
   chatMessages: initialChatMessages,
   isWaitingForChatResponse: false,
   chatMessageError: false,
 };
 
-export interface ChatContext {
-  levelId?: number;
-  systemPrompt: string;
-  studentCode: string;
-}
-
 // THUNKS
 
 // Compilation & Validation
 export const askAITutor = createAsyncThunk(
   'aitutor/askAITutor',
-  async (ChatContext: ChatContext, thunkAPI) => {
-    if (ChatContext.systemPrompt === undefined) {
+  async (chatContext: ValidationCompilationContext, thunkAPI) => {
+    const state = thunkAPI.getState() as {aiTutor: AITutorState};
+    const levelContext = {
+      levelId: state.aiTutor.level?.id,
+      isProjectBacked: state.aiTutor.level?.isProjectBacked,
+      scriptId: state.aiTutor.scriptId,
+    }
```

Comment:
@molly-moen like this? pull the level data directly from state instead of passing it to the tutor components?

Reply:
yep!